pkg/generators/cluster.go

```diff
@@ -0,0 +1,30 @@
+package generators
+
+import (
+	"fmt"
+	argoprojiov1alpha1 "github.com/argoproj-labs/applicationset/api/v1alpha1"
+	argov1alpha1 "github.com/argoproj/argo-cd/pkg/apis/application/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ Generator = (*ClusterGenerator)(nil)
+
+// ClusterGenerator generates Applications for some or all clusters registered with ArgoCD.
+type ClusterGenerator struct {
+	commonClient client.Client
+}
+
+func NewClusterGenerator(client client.Client) Generator {
```

Comment:
Instead of a client, can we accept an informer? This way we can avoid API calls and get the values from the informer cache.

Reply:
I believe the controller-runtime client is a delegating client and an abstraction over the informers, so reads via the client are not hitting the API server. (ref https://github.com/kubernetes-sigs/controller-runtime/issues/498) Does that seem ok? If not we can go informer. I think I'll actually just remove it for now as I'm not totally sure how to get access to the informers underneath but will handle that once I get going on impl if that is the preferred direction.

Reply:
I think the  controller-runtime client is fine

Reply:
Thanks for clarifying. I think you're right. Haven't used kubebuilder before so TIL.